solr/solrj-zookeeper/src/java/org/apache/solr/common/cloud/SolrZkClient.java

```diff
@@ -433,6 +435,14 @@ public byte[] getData(
     } else {
       result = keeper.getData(path, wrapWatcher(watcher), stat);
     }
+    if (CompressionUtil.isCompressedBytes(result)) {
+      log.debug("Zookeeper data at path {} is compressed", path);
+      try {
+        result = CompressionUtil.decompressBytes(result);
+      } catch (DataFormatException e) {
+        throw new RuntimeException(e);
```

Comment:
Please don't introduce random RuntimeExceptions that have no additional context. You can probably use a SolrException and explain the context where the decompression was happening.